posthog/temporal/data_imports/util.py

```diff
@@ -29,8 +30,11 @@ def prepare_s3_files_for_querying(
 
     s3_folder_for_querying = f"{s3_folder_for_job}/{normalized_table_name}__query"
 
-    if s3.exists(s3_folder_for_querying):
-        s3.delete(s3_folder_for_querying, recursive=True)
+    # TODO - maybe we should move these to a separate place and then delete them after the copy is done, to avoid any
+    # downtime?
+    if delete_existing:
+        if s3.exists(s3_folder_for_querying):
+            s3.delete(s3_folder_for_querying, recursive=True)
```

Comment:
We're getting to a point where this function should probably have some unit tests